pkg/cluster/cluster.go

```diff
@@ -138,6 +138,16 @@ func (c *Cluster) prepareBackupAndRestore() error {
 		}
 	}
 
+	// We use backup RS to mark that we have done backup creation and even restore process before.
```

Comment:
The backup == nil case is that we don't need to create backup RS. If so, we don't even have backup RS, and we do almost nothing in this func.